yt/utilities/lib/particle_kdtree_tools.pyx

```diff
@@ -29,19 +29,38 @@ from yt.utilities.lib.bounded_priority_queue cimport BoundedPriorityQueue
 
 cdef int CHUNKSIZE = 4096
 
+cdef struct axes_range:
+    int start
+    int stop
+    int step
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+cdef int set_axes_range(axes_range *axes, skipaxis):
+    axes.start = 0
+    axes.stop = 3
+    axes.step = 1
+    if skipaxis == 0:
+        axes.start = 1
+    if skipaxis == 1:
+        axes.step = 2
+    if skipaxis == 2:
+        axes.stop = 2
+    return 0
+
```

Comment:
this a fast way of allowing us to use the kdtree to calculate distances in dimensions less than 3.

I.e say we want to know the x and z distance for a y projection plot (not yet implemented), what is the fastest way to do this?
well putting an if statement in the for loop was too slow, and using a range(start, stop, step) also become slow because when the step is not hardcode cython generates tonnes of python code to deal with a negative step.
I have almost no change in performance by going it this way and setting up a while loop. If we need to skip a single axis this provides a very efficient means of doing so

Reply:
Nice trick! This is a nifty way to avoid branching in cython code.

Reply:
Cheers - I was pretty happy with the performance from this. It took a bit of prototyping to reach this solution